### DISCUSSION TITLE
启用oidc没有用

### DISCUSSION THREAD

**Discussion by voice-317418365** [hhyo/Archery#2377] [Q&A]
![image](https://github.com/hhyo/Archery/assets/35912159/c4a4bffe-fc4f-4e34-9fed-9a81cbab6ee9)
![image](https://github.com/hhyo/Archery/assets/35912159/aa8b7b0f-067b-49cd-bd5e-0e466bb438b8)
能帮忙看下配置有没有问题吗

**Comment by voice-317418365** [comment]
版本1.10，使用的docker-compose搭建的

**Comment by LeoQuote** [comment]
改环境变量，或者 .env 文件，看不懂的话就不要改settings.py

**Reply by voice-317418365** [reply]
![image](https://github.com/hhyo/Archery/assets/35912159/faeed962-b299-4dda-9cba-e269a44af6d3)
我修改.env文件，容器无法启动

**Reply by joenzx** [reply]
你最后改好了没？

**Reply by joenzx** [reply]
![image](https://github.com/hhyo/Archery/assets/42016493/f8b656dc-d51c-4d7b-a7fd-1add00d8c1d9)

![image](https://github.com/hhyo/Archery/assets/42016493/956a9d43-a56a-4d05-81f5-f5d47b39d562)
![image](https://github.com/hhyo/Archery/assets/42016493/9aa2d196-2416-4190-beeb-622fa8b30641)

我想集成飞书单点登录，这样添加了配置，从页面跳转时出现了这个问题，像是回调地址问题，能看下啥问题不，日志也没啥输出

**Reply by joenzx** [reply]
@LeoQuote

**Reply by arthas3014** [reply]
你好我想问下您配置成功了吗

**Reply by arthas3014** [reply]
@LeoQuote 我也遇到了 Code: mismatch_redirect_uri 这个问题

**Reply by LeoQuote** [reply]
实际的 redirect uri 是什么？

**Reply by arthas3014** [reply]
经过nginx代理redirect uri从https变为了http了，跟飞书配置的https不同了，但是我已经在nginx的配置里面加上了
`proxy_set_header   X-Forwarded-Proto $http_X_Forwarded_Proto`
结果代理的请求redirect uri还是变成http，很奇怪

**Reply by LeoQuote** [reply]
https://github.com/hhyo/Archery/discussions/2393#discussioncomment-7597432 请参考这里，删除 docker-compose 中的 NGINX_PORT 环境变量，如有更好的方案欢迎pr或更新文档

**Comment by arthas3014** [comment]
> [#2393 (comment)](https://github.com/hhyo/Archery/discussions/2393#discussioncomment-7597432) 请参考这里，删除 docker-compose 中的 NGINX_PORT 环境变量，如有更好的方案欢迎pr或更新文档

这个我已经删除掉了，可nginx代理过去还是http

**Reply by LeoQuote** [reply]
没太多能帮你的，我这边是一样的配置，也是开了https ，没有遇到你的问题，用的是mozilla django oidc ，你可以搜一下相关方案，或者debug 一下看看 https://github.com/mozilla/mozilla-django-oidc/issues/417

**Reply by arthas3014** [reply]
请问你这边起的ng，是项目里面docker目录下自带的那个ng吗，我用的是docker起的ng，可能会是这个原因？

**Reply by LeoQuote** [reply]
不太可能是你说的原因，nginx 都在用的，容器里的nginx 和外面的nginx 我都在用，你试一下我提供的贴子里的方案
